File: 1838_Frequency of the Most Frequent Element/FrequencyoftheMostFrequentElement.cs
public class Solution {
    public int MaxFrequency(int[] nums, int k) {
        Array.Sort(nums);
        
        var l = 0;
        var ans = 0;
        long sum = 0;
        for(var r=0;r < nums.Length;r++)
        {
            sum+=nums[r];
            while(l < r && (sum + k) < nums[r] * (r - l + 1))
                sum -= nums[l++];
            ans = Math.Max(ans, r - l + 1);
        }
        
        return ans;
    }
}

File: 1838_Frequency of the Most Frequent Element/FrequencyoftheMostFrequentElement.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
        Arrays.sort(nums);
        int l = 0;
        int ans = 0;
        long sum = 0;
        
        for(int r = 0;r<nums.length;r++) {
            sum += nums[r];
            while(l < r && ((sum + k) < nums[r] * (r - l+1)))
                sum -= nums[l++];    
            ans = Math.max(ans, r - l + 1);
        }
        
        return ans;
    }
}